Restructuring of business partner kit documentation

SujitMBRDI opened the pull request.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_api-reference.md:
```diff
@@ -0,0 +1,45 @@
+
+A gate for a member to share business partner data with CatenaX.
+
+Swagger: [https://partners-gate.int.demo.catena-x.net/ui/swagger-ui/index.html](https://partners-gate.int.demo.catena-x.net/ui/swagger-ui/index.html)
```
Don't provide links to consortia deployments. This guideline should be general for people who only have access to the open source repository of the BPDM project.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_api-reference.md:
```diff
@@ -0,0 +1,45 @@
+1. Site Controller
+2. legal Entity Controller
+3. Address Controller
+4. Business Partner Controller
```
Listing the controllers that are already listed in the Swagger/Open-API doesn't help much. Better add an explanation to what these Controllers are responsible for.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_api-reference.md:
```diff
@@ -0,0 +1,45 @@
+
+The Business Partner Pool API offers access to business partner data with for other Catena-X services.
+
+Swagger: [https://partners-pool.int.demo.catena-x.net/ui/swagger-ui/index.html](https://partners-pool.int.demo.catena-x.net/ui/swagger-ui/index.html)
```
Link to consortia deployment (see above)

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_api-reference.md:
```diff
@@ -0,0 +1,45 @@
+4. Meta Data Controller
+5. Business Partner Controller
+6. BPN Controller
+7. Adress Controller
```
Explanation of controller responsibilities needed

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_best-practices.md:
```diff
@@ -0,0 +1,19 @@
+description: ''
+sidebar_position: 4
+---
+1. Before using the BPDM services, ensure that you have the necessary authentication credentials and permissions to access the services. This may include obtaining an API key or token, and configuring your system to use it.
```
Maybe reference OAuth2 authentication flow at this place. The BPDM service currently does not accept API Keys and relies on an oAuth2 resource server to authenticate bearer tokens

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_documentation-bpdm.md:
```diff
@@ -0,0 +1,61 @@
+
+Getting started is easy. Here is a summary of the information that might help you get started:
+
+You will need to get a technical user and authentication via OAuth 2.0. You can find the getting started documentation with a description on how to get the technical user here. Once you have technical user, then you can get secrets from [vault](https://centralidp.demo.catena-x.net/) and can be used for authentication.
```
Don't refer to consortia vault here. Explanation needs to be more generic, as Vault and the centralidp are current consortia deployments that are not part of the eclipse-tractusx open source.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_documentation-bpdm.md:
```diff
@@ -0,0 +1,61 @@
+
+Token URL
+
+[https://centralidp.demo.catena-x.net/auth/realms/CX-Central/protocol/openidconnect/token](https://centralidp.demo.catena-x.net/auth/realms/CX-Central/protocol/openidconnect/token)
```
Links and explanation here should also be more generic to describe a non-sepcific oAuth2 server's urls.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_documentation-bpdm.md:
```diff
@@ -0,0 +1,61 @@
+
+#### Note
+
+It is important to follow the instructions mentioned in the respective files to ensure successful installation and configuration.
```
Reference files here from the open source repository. INSTALL.md and README.md both have a lot of information you detail here. Should include references to those.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_documentation-bpdm.md:
```diff
@@ -0,0 +1,61 @@
+5. Default configuration: BPDM Pool instance expected at localhost with port 8080.
+6. Configuration files: application.properties, application-auth.properties, and application-pool-auth.properties.
+7. REST API documentation: <http://localhost:8081/api/swagger-ui>.
+8. Profiles: default, auth, and auth-pool.
```
The Kit here needs to distinguish more clearly between two different use cases: A person who wants to run the BPDM service (either locally or deployed somewhere) and a person who wants to integrate against a deployed BPDM API. The INSTALL.md covers a lot of the first use case. The second use case is not covered well in the repository and should be the focus here in the Kit. 

The first part talks about how someone can authenticate to the API and then here it goes on what is required to install/deploy the API. Better separate and focus clearly on the 'usage' use case.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_troubleshooting.md:
```diff
@@ -0,0 +1,21 @@
+
+2. Authentication errors: These errors occur when the user is not authorized to access the service. To resolve this issue, ensure that the user has the appropriate permissions and that the authentication credentials are correct.
+
+    Example - If a user tries to use api services provided without authentication in, the server return an error message that states "401 Unauthorized."
```
Unauthorized occurs when the user is unknown, e.g. he didn't provide a valid Bearer Token. Permission problem is a 403 which should be distinguished here.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_troubleshooting.md:
```diff
@@ -0,0 +1,21 @@
+
+    Example - If a user tries to use api services provided without authentication in, the server return an error message that states "401 Unauthorized."
+
+3. Server errors: These errors occur when there is a problem with the server. To resolve this issue, check the server logs and contact the server administrator.
```
Also here sections should be divided into whether you are an admin or just use an API. In one case you can check the server logs and in the other case you need to contact an admin. Separate those use cases and provide troubleshooting advice for each separately

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
```diff
@@ -0,0 +1,2294 @@
+
+Here we have provided detailed examples of how the BPDM services can be used. This includes step-by-step instructions for each use case and code snippets showing how to make the API calls for better performance.
+
+## Use Case 1. Request Legal Entity by DE_VAT (via Legal Entity Controller)
```
DE_VAT is concrete example technical key for the German VAT identification number as you mentioned in the section. Make this more a general use case description like "Request Legal Entity by Identifier Number" or similar.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
````diff
@@ -0,0 +1,2294 @@
+Request to the Catena-X API for legal entities with a filter on EU VAT ID DE equals DE281308185. The response you received is a JSON object that contains information about the legal entities that match your query.
+
+```bash
+curl --location 'https://partners-pool.int.demo.catena-x.net/api/catena/legal-entities?EU_VAT_ID_DE=DE216038746' \
````
Again, no reference to consortia deployments

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
```diff
@@ -0,0 +1,2294 @@
+            }
+        }
+    ]
+}
```
I think the other can get the idea with explicitely showing one or two legal entity data. Just cut the response short by providing "..." in the content list.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
````diff
@@ -0,0 +1,2294 @@
+* Once you have leagal enity shown then direct response you can get using idType as BPN value as shown in below request and response.
+
+```bash
+curl --location 'https://partners-pool.int.demo.catena-x.net/api/catena/legal-entities/BPNL00000006M6CI?idType=BPN' \
````
consortia URL, see above

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
```diff
@@ -0,0 +1,2294 @@
+* The Legal Entity Controller sends an API call to the internal API with the EU_VAT_ID_DE.
+* The internal API returns information about only one legal entity.
+* The Legal Entity Controller displays the information about the legal entity to the user.
+* Once you have leagal enity shown then direct response you can get using idType as BPN value as shown in below request and response.
```
Flow or steps are good but internal system steps should be omitted. The steps should be from the perspective of the user. What does the user need to do and what does he see as a (intermediate) result.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
````diff
@@ -0,0 +1,2294 @@
+        }
+    ]
+}
+```
````
Also here, shorten the return call, so the user can get the idea without too much information overload

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
```diff
@@ -0,0 +1,2294 @@
+            }
+        }
+    ]
+}
```
shorten respsone (see above)

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
```diff
@@ -0,0 +1,2294 @@
+* The Legal Entity Controller sends an API call to the internal API with the EU_VAT_ID_DE.
+* The internal API returns information about the legal entity.
+* The Legal Entity Controller reads the BPNL (Business Partner Number List) from the API response.
+* The Legal Entity Controller displays the BPNL for the legal entity to the user e.g. "bpn": "BPNL00000006M6CI".
```
What kind of use case is that? Feels like that it exceeds use case 1 in which you only want to request a legal entity by identifier number

SujitMBRDI replied:
Yes, it is part of use case 1. May be i shall show you the page then it will be more clear or will omit this heading.

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
````diff
@@ -0,0 +1,2294 @@
+        }
+    ]
+}
+```
````
shorten response (see above)

nicoprow commented on docs/kits/Business Partner Kit/Documentation BPDM/page_use-cases.md:
```diff
@@ -0,0 +1,2294 @@
+
+### 1.4 Fetch sites for legal entity via API
+
+Scenario: The user enters a valid BPN under pathparam and the internal API returns information about site legal entities.
```
That's a different use case and should be presented as such

nicoprow reviewed (changes requested):
Feedback in the comments.

In general, we also need valid example business partners to show here. The business partners here are too concrete for my taste and should be curated.

maximilianong commented:
@SujitMBRDI you will update this PR with your latest branch right?

SujitMBRDI commented:
> @SujitMBRDI you will update this PR with your latest branch right?

@maximilianong, Yes, i have already updated the changes which i showed you till now.
Changes related to openApi and use case we discussed, that i shall push by tomorrow EOD.

maximilianong commented:
> > @SujitMBRDI you will update this PR with your latest branch right?
> 
> @maximilianong, Yes, i have already updated the changes which i showed you till now. Changes related to openApi and use case we discussed, that i shall push by tomorrow EOD.

Ok - we will review it with @nicoprow next week before Nico or I merge it.

SujitMBRDI commented:
> > > @SujitMBRDI you will update this PR with your latest branch right?
> > 
> > 
> > @maximilianong, Yes, i have already updated the changes which i showed you till now. Changes related to openApi and use case we discussed, that i shall push by tomorrow EOD.
> 
> Ok - we will review it with @nicoprow next week before Nico or I merge it.

@maximilianong, I have committed changes that we discussed related to test data setup use case . Also added open api for gate service with api reference page updation. Kindly review the changes. I am not able to find option to add you as reviewer so @nicoprow is the only reviewer for this PR

SujitMBRDI commented:
> Feedback in the comments.
> 
> In general, we also need valid example business partners to show here. The business partners here are too concrete for my taste and should be curated.

@nicoprow, I have covered all topics we discussed and also resolved conflicts i got as PR is pending from long time.
Clustering on open api for both Gate and Pool api is also added now it is grouped with tag names of every path.

SujitMBRDI commented:
@maximilianong and @nicoprow, Kindly review and merge change request. Here,  i have added clustering, enhanced structure for documentation on business partner kit also provided different view for Development View doc. 

If we keep pull request open for long time, then i am getting multiple conflicts. Currently, i have resolve conflicts and pull request is good to merge.
